src/main/java/org/zalando/nakadi/service/FloodService.java

```diff
@@ -0,0 +1,199 @@
+                        final SubscriptionDbRepository subscriptionDbRepository,
+                        final ZooKeeperHolder zooKeeperHolder,
+                        final NakadiSettings nakadiSettings) {
+        this.eventTypeCache = eventTypeCache;
+        this.zooKeeperHolder = zooKeeperHolder;
+        this.subscriptionDbRepository = subscriptionDbRepository;
+        this.retryAfterStr = String.valueOf(nakadiSettings.getRetryAfter());
+    }
+
+    @PostConstruct
+    public void initIt() {
+        try {
+            this.floodersCache = TreeCache.newBuilder(zooKeeperHolder.get(), PATH_FLOODER).setCacheData(false).build();
+            this.floodersCache.start();
+        } catch (final Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
+
+    @PreDestroy
+    public void cleanUp() {
+        this.floodersCache.close();
+    }
+
+    public boolean isBlocked(final Type type, final String name) {
+        try {
+            final boolean blocked = floodersCache.getCurrentData(type + "/" + name) != null;
+            if (blocked) {
+                LOG.info("{} {} is blocked", type.name(), name);
+            }
+            return blocked;
+        } catch (final Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
+        return false;
+    }
+
+    public boolean isProductionBlocked(final String etName, final String appId) {
+        return isBlocked(Type.PRODUCER_ET, etName) || isBlocked(Type.PRODUCER_APP, appId);
+    }
+
+    public boolean isConsumptionBlocked(final String etName, final String appId) {
+        return isBlocked(Type.CONSUMER_ET, etName) || isBlocked(Type.CONSUMER_APP, appId);
+    }
+
+    public boolean isSubscriptionConsumptionBlocked(final String subscriptionId, final String appId) {
+        try {
+            return isSubscriptionConsumptionBlocked(
+                    subscriptionDbRepository.getSubscription(subscriptionId).getEventTypes(), appId);
+        } catch (final NakadiException e) {
+            LOG.error(e.getMessage(), e);
+        }
+        return false;
+    }
+
+    public boolean isSubscriptionConsumptionBlocked(final Collection<String> etNames, final String appId) {
+            return etNames.stream()
+                    .map(etName -> isBlocked(Type.CONSUMER_ET, etName)).findFirst().orElse(false) ||
+                    isBlocked(Type.CONSUMER_APP, appId);
+    }
+
+    public Map<String, Map> getFlooders() {
+        return new HashedMap() {
+            {
+                put("consumers", new HashMap<String, Set<String>>() {{
+                    put("event_types", getChildren(Type.CONSUMER_ET));
+                    put("apps", getChildren(Type.CONSUMER_APP));
+                }});
+                put("producers", new HashMap<String, Set<String>>() {{
+                    put("event_types", getChildren(Type.PRODUCER_ET));
+                    put("apps", getChildren(Type.PRODUCER_APP));
+                }});
```

Comment:
@antban
1. there was no intention to call this method 100 times per second.
2. i did not know about that problem, thanks!
fixed 1af2a29